### PR TITLE
feat(doctor): orchestrator + finding contract; extract memory check

### PR DESCRIPTION
**Pull request by howarewoo**
## Goal

Increment 2/7 of /woostack-doctor: turn the moved engine into a check **orchestrator** with a stable finding contract and CI-friendly exit codes, so later increments add checks as `checks/*.sh`.

## Summary

- `doctor.sh` now runs `checks/*.sh`, groups findings (`severity⇥code⇥fixable⇥path⇥message`), prints `::error::`/`::warning::` annotations, dumps machine findings on stdout, exits nonzero iff any `error`. `--check` = CI mode (annotations + exit only).
- Memory lint extracted verbatim into `checks/memory.sh` (severities preserved; all `fixable=report`).
- `test-doctor.sh` reworked to the workspace-root contract; new `test-orchestrator.sh` (missing-`.woostack`→exit 2, clean→0, warn-only→0, error→1, `--check` suppresses dump, unknown flag→2).

### Automated

- [x] `run-tests.sh` → test-doctor 47, test-orchestrator 8, test-no-stale-paths 1 — all pass.
- [x] init suite still green; orchestrator on the real store exits 0.

### Manual

**Before merge**

- [ ] Confirm `checks/memory.sh` preserves the original lint flags + severities (diff vs the pre-move `doctor.sh`).

Spec: .woostack/specs/2026-06-13-woostack-doctor.md

### CODE REVIEW

**Comment by howarewoo** [comment]
* **#343** <a href="https://app.graphite.com/github/pr/howarewoo/woostack/343?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#342** <a href="https://app.graphite.com/github/pr/howarewoo/woostack/342?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#341** <a href="https://app.graphite.com/github/pr/howarewoo/woostack/341?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#340** <a href="https://app.graphite.com/github/pr/howarewoo/woostack/340?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#339** <a href="https://app.graphite.com/github/pr/howarewoo/woostack/339?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#338** <a href="https://app.graphite.com/github/pr/howarewoo/woostack/338?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#337** <a href="https://app.graphite.com/github/pr/howarewoo/woostack/337?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/howarewoo/woostack/337?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#336** <a href="https://app.graphite.com/github/pr/howarewoo/woostack/336?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#335** <a href="https://app.graphite.com/github/pr/howarewoo/woostack/335?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: skills/woostack-doctor/scripts/tests/test-orchestrator.sh [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+warnws="$(mktemp -d)"; mkdir -p "$warnws/.woostack/memory"
+printf -- '---\nname: n\ntype: gotcha\nscope: *\nsource: .woostack/specs/x.md\nupdated: 2099-01-01\n---\nbody [[ghost]]\n' > "$warnws/.woostack/memory/n.md"
+bash "$DOC" "$warnws" >/dev/null 2>&1; assert_exit 0 "$?" "warn-only exits 0"
+
+errws="$(mktemp -d)"; mkdir -p "$errws/.woostack/memory"
+printf 'no fence\n' > "$errws/.woostack/memory/bad.md"
+bash "$DOC" "$errws" >/dev/null 2>&1; assert_exit 1 "$?" "error finding exits 1"
+
+dump="$(bash "$DOC" --check "$warnws" 2>/dev/null)"
+assert_eq "$dump" "" "--check suppresses machine dump on stdout"
```
Comment (howarewoo) [inline skills/woostack-doctor/scripts/tests/test-orchestrator.sh:25]: **Nit: No test: --check + error workspace exits 1**

The orchestrator test verifies --check suppresses stdout on a warn-only workspace, but never asserts that --check with an error-producing workspace still exits 1. The CI contract of --check is that it gates on exit code; a consumer running `doctor.sh --check .` in CI needs confidence the exit code reflects errors even in check mode. A future regression (e.g., CHECK_ONLY path swallowing the error counter) would go undetected.

Fix: Add a test after line 26 (the assert_eq for --check dump): `bash "$DOC" --check "$errws" >/dev/null 2>&1; assert_exit 1 "$?" "--check with errors still exits 1"`

```suggestion
bash "$DOC" --check "$errws" >/dev/null 2>&1
assert_exit 1 "$?" "--check with errors still exits 1"
```

<sub>— <strong>MEDIUM · NIT</strong> · flagged by the <code>tests</code> agent</sub>

**Review by howarewoo — COMMENTED** [review]
## AI Deep Code Review Summary

Swarm ran 4 angles (bugs, security, conventions, tests) over the orchestrator + finding-contract + memory-check extraction (~352 LOC). One non-blocking nit on test coverage; the orchestrator and check contract are otherwise sound.

---
**Status: APPROVED** — No blocking findings, 1 nit(s). See inline comments.
*Audited by woostack-review · Host: claude-code · Provider: anthropic · Model: claude-opus-4-8 (validators) + claude-sonnet-4-6 (angles)*



_Review event downgraded to COMMENT — GitHub blocks APPROVE on your own PR. Status line above carries the actual verdict._

File: skills/woostack-doctor/scripts/tests/test-orchestrator.sh (hunk 1 above)
Comment (howarewoo) [reply skills/woostack-doctor/scripts/tests/test-orchestrator.sh:25]: Fixed in ae48a56 — added `bash "$DOC" --check "$errws" >/dev/null 2>&1; assert_exit 1 "$?"` so the suite now pins the CI-gating exit code in `--check` mode (was only asserted in default mode). Suite: 9 passed, 0 failed.

**Review by howarewoo — COMMENTED** [review]
## AI Deep Code Review Summary — re-review (round 2)

Re-ran the full swarm after the test fix (commit `ae48a56`). The added `--check`+errors exit-1 assertion is present and the suite is green (9 passed). The adversarial validators surfaced no agreed findings on the full diff; the prior nit thread is resolved.

---
**Status: APPROVED** — Clean. No blocking findings, no open threads.
*Audited by woostack-review · Host: claude-code · Provider: anthropic · Model: claude-opus-4-8 (validators) + claude-sonnet-4-6 (angles)*

_Review event downgraded to COMMENT — GitHub blocks APPROVE on your own PR. Status line above carries the actual verdict._

**Comment by howarewoo** [comment]
### Merge activity

* **Jun 14, 3:35 AM UTC**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.com/github/pr/howarewoo/woostack/337).
* **Jun 14, 3:36 AM UTC**: [Graphite](https://app.graphite.com/github/pr/howarewoo/woostack/337) rebased this pull request as part of a merge.
* **Jun 14, 3:37 AM UTC**: @howarewoo merged this pull request with [Graphite](https://app.graphite.com/github/pr/howarewoo/woostack/337).
